test/jpypetest/test_number.py

```diff
@@ -0,0 +1,40 @@
+import sys
+import jpype
+import common
+import random
+import _jpype
+import jpype
+from jpype import java
+from jpype.types import *
+try:
+    import numpy as np
+except ImportError:
+    pass
+
+class JNumberTestCase(common.JPypeTestCase):
+    def setUp(self):
+        common.JPypeTestCase.setUp(self)
+        self.cls = JClass("jpype.common.Fixture")
+        self.fixture = self.cls()
+
+    def testJavaPrimitives(self):
+        self.assertIsInstance(self.fixture.callNumber(JByte(1)), java.lang.Byte)
+        self.assertIsInstance(self.fixture.callNumber(JShort(1)), java.lang.Short)
```

Comment:
just for the sake of completeness, we seem to lack JCharacter?

Reply:
Characters are not numbers as they have encodings.  There are places where we treat them as unsigned shorts but that is mostly incorrect.   So as you can see from the language definition neither Boolean nor Character are derived from Number

https://docs.oracle.com/javase/8/docs/api/java/lang/Number.html

Reply:
I missed that important difference.